File: face.py
import cv2
import numpy


class Face:

    def __init__(self):
        self.face_detector = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_frontalface_default.xml')

    def locations(self, image: numpy.ndarray) -> []:
        if len(image.shape) == 3:
            image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
        image = cv2.equalizeHist(image)
        return self.face_detector.detectMultiScale(image,
                                                   scaleFactor=1.1,
                                                   minNeighbors=3,
                                                   minSize=(32, 32))

File: C++Version/face/face.cpp
#include "face.h"

Face::Face(const string faceModel) { faceDetector.load(faceModel); }

// Face Detection
void Face::locations(const Mat &image, vector<Rect> &faces) {
    Mat gray = image.clone();
    if (gray.channels() > 1)
        cvtColor(gray, gray, COLOR_BGR2GRAY);
    equalizeHist(gray, gray);
    faceDetector.detectMultiScale(gray, faces, 1.1, 3, 0, Size(32, 32));
}
